File: abc090/abc090_d.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
  int n,k;
  cin >> n >> k;

  long long ans=0;
  if(k==0) ans=(long long) n*n;
  else {
    for (int b = k+1; b <= n; b++)
    {
      int cnt = (n+1)/b*(b-k)+ max(0,(n+1)%b-k);
      ans+=(long long) cnt;
    }
  }
  cout << ans << endl;  
}

File: abc090/abc090_d.rb
n,k=gets.split.map(&:to_i)
ans=n*n
if(k!=0)
  cnt=0
  (k+1).upto(n) do |b|
    cnt+=(n+1)/b*(b-k)+ [0,(n+1)%b-k].max
  end
  ans=cnt
end
puts ans